grails-app/services/org/pih/warehouse/data/LoadDataService.groovy

```diff
@@ -0,0 +1,187 @@
+/**
+* Copyright (c) 2012 Partners In Health.  All rights reserved.
+* The use and distribution terms for this software are covered by the
+* Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+* which can be found in the file epl-v10.html at the root of this distribution.
+* By using this software in any fashion, you are agreeing to be bound by
+* the terms of this license.
+* You must not remove this notice, or any other, from this software.
+**/
+package org.pih.warehouse.data
+
+
+import org.grails.plugins.csv.CSVMapReader
+import org.pih.warehouse.core.*
+import org.pih.warehouse.importer.ImportDataCommand
+import org.pih.warehouse.product.ProductCatalog
+
+class LoadDataService {
```

Comment:
I think these methods should go into DataService.

Reply:
I do not think it is possible. LoadDataService requires productService, which requires dataService. After moving to DataService there is creating bean exception, because of circular reference between DataService and ProductService